src/main/java/bio/terra/cli/service/utils/WorkspaceManagerService.java

```diff
@@ -332,4 +365,297 @@ public RoleBindingList getRoles(UUID workspaceId) {
+                    .location(location));
+    try {
+      return controlledGcpResourceApi
+          .createBigQueryDataset(createRequest, workspaceId)
+          .getBigQueryDataset();
+    } catch (ApiException ex) {
+      throw new SystemException(
+          "Error creating controlled Big Query dataset in the workspace.", ex);
+    }
+  }
+
+  /**
+   * Call the Workspace Manager DELETE
+   * "/api/workspaces/v1/{workspaceId}/resources/referenced/gcp/buckets/{resourceId}" endpoint to
+   * delete a GCS bucket as a referenced resource in the workspace.
+   *
+   * @param workspaceId the workspace to remove the resource from
+   * @param resourceId the resource id
+   */
+  public void deleteReferencedGcsBucket(UUID workspaceId, UUID resourceId) {
+    ReferencedGcpResourceApi referencedGcpResourceApi = new ReferencedGcpResourceApi(apiClient);
+    try {
+      referencedGcpResourceApi.deleteBucketReference(workspaceId, resourceId);
+    } catch (ApiException ex) {
+      throw new SystemException("Error deleting referenced GCS bucket in the workspace.", ex);
+    }
+  }
+
+  /**
+   * Call the Workspace Manager DELETE
+   * "/api/workspaces/v1/{workspaceId}/resources/referenced/gcp/bigquerydatasets/{resourceId}"
+   * endpoint to delete a Big Query dataset as a referenced resource in the workspace.
+   *
+   * @param workspaceId the workspace to remove the resource from
+   * @param resourceId the resource id
+   */
+  public void deleteReferencedBigQueryDataset(UUID workspaceId, UUID resourceId) {
+    ReferencedGcpResourceApi referencedGcpResourceApi = new ReferencedGcpResourceApi(apiClient);
+    try {
+      referencedGcpResourceApi.deleteBigQueryDatasetReference(workspaceId, resourceId);
+    } catch (ApiException ex) {
+      throw new SystemException(
+          "Error deleting referenced Big Query dataset in the workspace.", ex);
+    }
+  }
+
+  /**
+   * Call the Workspace Manager POST
+   * "/api/workspaces/v1/{workspaceId}/resources/referenced/gcp/buckets/{resourceId}" endpoint to
+   * delete a GCS bucket as a controlled resource in the workspace.
+   *
+   * @param workspaceId the workspace to remove the resource from
+   * @param resourceId the resource id
+   */
+  public void deleteControlledGcsBucket(UUID workspaceId, UUID resourceId) {
+    ControlledGcpResourceApi controlledGcpResourceApi = new ControlledGcpResourceApi(apiClient);
+    String asyncJobId = UUID.randomUUID().toString();
+    DeleteControlledGcpGcsBucketRequest deleteRequest =
+        new DeleteControlledGcpGcsBucketRequest().jobControl(new JobControl().id(asyncJobId));
+    // TODO (PF-719): factor out this polling pattern into a utility method
+    try {
+      DeleteControlledGcpGcsBucketResult deleteResult =
+          controlledGcpResourceApi.deleteBucket(deleteRequest, workspaceId, resourceId);
+      while (deleteResult.getJobReport().getStatus().equals(JobReport.StatusEnum.RUNNING)) {
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
I don't think as implemented this completes anytime soon for non-empty buckets. I think for the CLI we might want to be able to handle operations that take a long time. But fine not to address in this PR.
@zloery

Reply:
Certainly it should time out at some point. I'm planning to handle this in a follow-on ticket (see TODO comment above). If this will often take a long time, then I'll make the timeout short and perhaps show a "check back later" type message instead of "failure" if it does time out.